tests/components/huawei_lte/test_switches.py

```diff
@@ -133,6 +113,38 @@ async def test_turn_off_switch_wifi_guest_network(
     )
     await hass.async_block_till_done()
     assert hass.states.is_state(SWITCH_WIFI_GUEST_NETWORK, STATE_OFF)
-    hass.data[DOMAIN].routers[
-        "test-serial-number"
-    ].client.wlan.wifi_guest_network_switch.assert_called_with(False)
+    client.return_value.wlan.wifi_guest_network_switch.assert_called_with(False)
+
+
+@patch("homeassistant.components.huawei_lte.Connection", MagicMock())
+@patch(
+    "homeassistant.components.huawei_lte.Client",
+    return_value=magic_client({"Ssids": {"Ssid": "str"}}),
+)
+async def test_huawei_lte_wifi_guest_network_config_entry_when_ssid_is_str(
+    client, hass: HomeAssistant
+):
+    """Test switch wifi guest network config entry when ssid is a str. Issue #76244. Huawai models: H312-371, E5372 and E8372."""
```

Comment:
The first line of the docstring is a header line and should only have one sentence. Further sentences can come in the docstring body separated by a blank line from the header line. Docstring lines should preferably not be longer than 72 characters according to PEP8.
```suggestion
    """Test switch wifi guest network config entry when ssid is a str.
    
    Issue #76244. Huawai models: H312-371, E5372 and E8372.
    """
```